tests/test_controller.py

```diff
@@ -765,3 +798,59 @@ def test_preload(self):
                 assert self.english_1.title not in response.data
                 assert self.english_2.title in response.data
                 assert self.french_1.author not in response.data
+
+
+class TestScopedSession(ControllerTest):
+    """Test that in production scenarios (as opposed to normal unit tests)
+    the app server runs each incoming request in a separate database
+    session.
+
+    Compare to TestBaseController.test_unscoped_session, which tests
+    the corresponding behavior in unit tests.
+    """
+
+    def setup(self):
+        from api.app import _db
+        super(TestScopedSession, self).setup(_db)
+
+    def test_scoped_session(self):
+        # Start a simulated request to the Flask app server.
+        with self.app.test_request_context("/"):
+            # Each request is given its own database session distinct
+            # from the one used by most unit tests or the one
+            # associated with the CirculationManager object.
+            session1 = current_session()
+            assert session1 != self._db
+            assert session1 != self.app.manager._db
+
+            # Add an Identifier to the database.
+            identifier = Identifier(type=DataSource.GUTENBERG, identifier="1024")
+            session1.add(identifier)
+            session1.flush()
+
+            # The Identifier immediately shows up in the session that
+            # created it.
+            [identifier] = session1.query(Identifier).all()
+            eq_("1024", identifier.identifier)
+
+            # It doesn't show up in self._db, the database session
+            # used by most other unit tests, because it was created
+            # within the (still-active) context of a Flask request,
+            # which happens within a nested database session.
+            eq_([], self._db.query(Identifier).all())
```

Comment:
It does show up in app.manager._db, and I'm pretty sure that's because app.manager._db is managing the parent transaction.